Dart/thirdPartySrc/analysisServer/com/google/dart/server/internal/remote/processor/DartfixProcessor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2019, the Dart project authors.
+ *
+ * Licensed under the Eclipse Public License v1.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.dart.server.internal.remote.processor;
+
+import com.google.dart.server.DartfixConsumer;
+import com.google.dart.server.FormatConsumer;
+import com.google.gson.JsonObject;
+
+import org.dartlang.analysis.server.protocol.DartFixSuggestion;
+import org.dartlang.analysis.server.protocol.RequestError;
+import org.dartlang.analysis.server.protocol.SourceEdit;
+import org.dartlang.analysis.server.protocol.SourceFileEdit;
+
+import java.util.List;
+
+/**
+ * Instances of {@code DartfixProcessor} translate JSON result objects for a given
+ * {@link DartfixConsumer}.
+ *
+ * @coverage dart.server.remote
+ */
+public class DartfixProcessor extends ResultProcessor {
+  private final DartfixConsumer consumer;
+
+  public DartfixProcessor(DartfixConsumer consumer) {
+    this.consumer = consumer;
+  }
+
+  public void process(JsonObject resultObject, RequestError requestError) {
+    if (resultObject != null) {
+      try {
+        List<DartFixSuggestion> suggestions = DartFixSuggestion.fromJsonArray(resultObject.get("suggestions").getAsJsonArray());
+        List<DartFixSuggestion> otherSuggestions = DartFixSuggestion.fromJsonArray(resultObject.get("otherSuggestions").getAsJsonArray());
+        boolean hasErrors = resultObject.get("hasErrors").getAsBoolean();
+        List<SourceFileEdit> edits = SourceFileEdit.fromJsonArray(resultObject.get("edits").getAsJsonArray());
```

Comment:
@jwren Would it be better if the server returned `SourceChange` instead of `List<SourceFileEdit>`?

Reply:
This will ultimately be up to the dart server devs, when they figure out the UX for the migration.

Reply:
Maybe. The protocol was originally designed for use by the command-line dartfix tool, and there wasn't any interest in using the other fields so we didn't include them. But if other clients want to make use of any or all of those fields, then we can certainly change the protocol.